rest_framework/relations.py

```diff
@@ -267,7 +267,9 @@ def to_representation(self, value):
                 'Could not resolve URL for hyperlinked relationship using '
                 'view name "%s". You may have failed to include the related '
                 'model in your API, or incorrectly configured the '
-                '`lookup_field` attribute on this field.'
+                '`lookup_field` attribute on this field, or perhaps your'
+                'data contains an empty value for the `lookup_field` you'
+                'specified.'
```

Comment:
Perhaps we could instead catch `value in ('', None)` and only issue a reworded message in that case?